tests/app/billing/test_billing.py

```diff
@@ -282,118 +284,90 @@ def test_create_free_sms_fragment_limit(client, sample_service):
 
-    assert response_get.status_code == 404
+    assert response.status_code == 201
+    assert response_get.status_code == 200
+    assert json_resp['financial_year_start'] == current_year
+    assert json_resp['free_sms_fragment_limit'] == 9999
+
 
+def test_get_free_sms_fragment_limit_current_year(client, sample_service):
 
-def test_get_free_sms_fragment_limit_unknown_service_id_return_404(client):
+    current_year = get_current_financial_year_start_year()
+    create_annual_billing(sample_service.id, 9999, current_year - 1)
 
     response_get = client.get(
-        'service/{}/billing/free-sms-fragment-limit'.format(uuid.uuid4()),
+        'service/{}/billing/free-sms-fragment-limit'.format(sample_service.id),
         headers=[('Content-Type', 'application/json'), create_authorization_header()])
+
     json_resp = json.loads(response_get.get_data(as_text=True))
-    assert response_get.status_code == 404
+    assert response_get.status_code == 200
+    assert json_resp['financial_year_start'] == get_current_financial_year_start_year()
+    assert json_resp['free_sms_fragment_limit'] == 250000
 
 
-def test_get_free_sms_fragment_limit_current_year(client, sample_service):
-    response = client.get(
-        'service/{}/billing/free-sms-fragment-limit/current-year'.format(sample_service.id, True),
-        headers=[('Content-Type', 'application/json'), create_authorization_header()])
-    json_resp = json.loads(response.get_data(as_text=True))
+def test_get_free_sms_fragment_limit_past_year_not_exist(client, sample_service):
+    current_year = get_current_financial_year_start_year()
+    create_annual_billing(sample_service.id, 9999, current_year - 1)
+    create_annual_billing(sample_service.id, 10000, current_year + 1)
 
-    assert response.status_code == 200
-    assert json_resp['data']['free_sms_fragment_limit'] == 250000
+    res_get = client.get(
+        'service/{}/billing/free-sms-fragment-limit?financial_year_start={}'
+        .format(sample_service.id, current_year - 2),
+        headers=[('Content-Type', 'application/json'), create_authorization_header()])
+    json_resp = json.loads(res_get.get_data(as_text=True))
 
+    assert res_get.status_code == 200
+    assert json_resp['financial_year_start'] == current_year - 1
+    assert json_resp['free_sms_fragment_limit'] == 9999
 
-def test_post_free_sms_fragment_limit_current_year(client, sample_service):
 
-    data_new = {'free_sms_fragment_limit': 7777}
-    response = client.post('service/{}/billing/free-sms-fragment-limit'.format(sample_service.id),
-                           data=json.dumps(data_new),
-                           headers=[('Content-Type', 'application/json'), create_authorization_header()])
+def test_get_free_sms_fragment_limit_future_year_not_exist(client, sample_service):
+    current_year = get_current_financial_year_start_year()
+    create_annual_billing(sample_service.id, 9999, current_year - 1)
```

Comment:
Same here, not sure if this shows anything, as you just want to check that there is a setting for years beyond the existing setting

Reply:
agreed. Changed.